packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -318,7 +341,8 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     genesisAccounts: GenesisAccount[],
     tracingConfig?: TracingConfig,
     private _forkNetworkId?: number,
-    nextBlockBaseFee?: BN
+    nextBlockBaseFee?: BN,
+    private readonly _hardforkActivationBlocks: HardforkActivationBlocks = mainnetHardforkActivations
```

Comment:
This doesn't make sense to default to mainnet because users will use other networks too (testnets, etc).